extensions/amp-carousel/0.2/test/test-type-slides.js

```diff
@@ -290,6 +290,19 @@ describes.realWin(
 
         expect(eventSpy).to.have.been.calledOnce;
       });
+
+      it('should dispatch event with index and actionTrust when changing slides', async () => {
+        container.addEventListener('slideChange', (event) => {
+          const {index, actionTrust} = event.data;
```

Comment:
Can you do one of these:
- Assert the event has been emitted the expected number of times after `afterIndexUpdate`, either with a flag or by [counting the events](https://github.com/ampproject/amphtml/blob/113aa44cce56cd236a54149eb375714dd4d4ae3b/extensions/amp-carousel/0.1/test/test-slidescroll.js#L1270)
- Give a [spy](https://sinonjs.org/releases/latest/spies/) as the event listener callback and count calls / assert arguments have this shape

Reply:
Also, does it work if the event listener is attached on the document, or only on the container?

Reply:
The event listener is attached on the container and because of that, I cannot count the event using `win.document.eventListeners.count('slideChange')` statement (doesn't work on the container element). Used the counter approach as discussed offline. Also I removed the `should dispatch when changing slides` test since this test is a superset of that.